src/operators/timeoutWith.ts

```diff
@@ -43,12 +43,12 @@ class TimeoutWithSubscriber<T, R> extends OuterSubscriber<T, R> {
     return this._hasCompleted;
   }
 
-  constructor(destination: Subscriber<T>,
+  constructor(public destination: Subscriber<T>,
               private absoluteTimeout: boolean,
               private waitFor: number,
               private withObservable: Observable<any>,
               private scheduler: Scheduler) {
-    super(destination);
```

Comment:
Ok, I'll look into that.
However, we use this same approach in many other operators. Either we add tests for keeping the subscription chain that you mentioned, or we change all those other operators too.